multik-jvm/src/test/kotlin/org/jetbrains/kotlinx/multik_jvm/linAlg/JvmLinAlgTest.kt

```diff
@@ -0,0 +1,254 @@
+//    }
+
+//    @Test
+//    fun `gemm test`() {
+//        val procedurePrecision = 1e-5
+//
+//        val a = mk.ndarray(mk[mk[1.0, 2.0, 3.0, 2.0, 1.0], mk[4.0, 5.0, 6.0, 5.0, 4.0]])
+//        val b = mk.ndarray(mk[mk[7.0, 8.0], mk[9.0, 10.0], mk[11.0, 12.0], mk[13.0, 14.0], mk[15.0, 16.0]])
+//        val c = mk.ndarray(mk[mk[11.0, 13.0], mk[15.0, 17.0]])
+//        val adotb = mk.ndarray(mk[mk[110.0, 121.0], mk[279.0, 305.0]])
+//
+//        //        proof: numpy
+//        //
+//        //        a = np.array([[1.0, 2.0, 3.0, 2.0, 1.0], [4.0, 5.0, 6.0, 5.0, 4.0]])
+//        //        b = np.array([[7.0, 8.0], [9.0, 10.0], [11.0, 12.0], [13.0, 14.0], [15.0, 16.0]])
+//        //        print(a @ b + c)
+//
+//        gemm(a, 0, 0, 2, 5, 1.0,
+//             b, 0, 0, 5, 2,
+//             c, 0, 0, 2, 2, 1.0)
+//
+//        assert(c == adotb)
+//
+//    }
+
+
+
+    @Test
+    fun `test plu`() {
+
+        val procedurePrecision = 1e-5
+
+        val iters = 10000
+        val sideFrom = 1
+        val sideUntil = 100
+        for (all in 0 until iters) {
+
+            val rnd = Random(System.currentTimeMillis())
+            val m = rnd.nextInt(sideFrom, sideUntil)
+            val n = rnd.nextInt(sideFrom, sideUntil)
+
+
+            val a = mk.d2array<Double>(m, n) { rnd.nextDouble() }
+
+            val (P, L, U) = PLU(a)
+
+            val diff = a - dot(P, dot(L, U))
+            val maxdiff = max(diff)
+            val mindiff = min(diff)
+
+            val ok = maxdiff < procedurePrecision && maxdiff - mindiff < procedurePrecision
+
+
+            if(!ok) {
+                println("wrong on:")
+                println(a)
+                println("PLU gives:")
+                println(P)
+                println(L)
+                println(U)
+                println("P*L*U = ")
+                println(dot(P, dot(L, U)))
+                assert(false)
+
+            }
+
+        }
+    }
+
+    @Test
+    fun `plu benchmark`() {
+        val rnd = Random(424242)
+        val iters = 1000
+        val maxSideLen = 400
+        var avgTime = 0.0
+        var maxTime = 0.0
+        var minTime = Double.POSITIVE_INFINITY
+
+
+        for (all in 0 until iters) {
+
+            val m = rnd.nextInt(maxSideLen / 2, maxSideLen + 1)
+            val n = rnd.nextInt(maxSideLen / 2, maxSideLen + 1)
+
+
+            val a = mk.d2array<Double>(m, n) { rnd.nextDouble() }
+
+            val newtime = measureTimeMillis {
+                val (P, L, U) = PLU(a)
+            }.toDouble()
+            maxTime = kotlin.math.max(maxTime, newtime)
+            minTime = kotlin.math.min(minTime, newtime)
+            avgTime += newtime
+        }
+        avgTime /= iters
+        println("On $iters iteration on random matrices with random side in range [${maxSideLen / 2}, $maxSideLen]:\n" +
+                "avgTime: $avgTime ms\nmaxTime: $maxTime ms\nminTime: $minTime ms")
+    }
```

Comment:
Such benchmarks are not so necessary. If you still want to leave them, then it is better to take them out separately.